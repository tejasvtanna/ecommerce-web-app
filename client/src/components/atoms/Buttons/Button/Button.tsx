import React from 'react'
import CSS from 'csstype'
import { Button as RButton } from 'react-bootstrap'

interface Props {
  variant?: string
  block?: boolean
  customStyle?: CSS.Properties
  disabled?: boolean
  type?: string
  onClick?: any
}

export const Button: React.FC<Props> = ({
  block = true,
  variant,
  disabled = false,
  customStyle,
  onClick,
  children,
  type,
}) => {
  return (
    <>
      {!variant && (
        <RButton
          style={{ backgroundColor: '#ff3e6c', borderColor: '#ff3e6c', ...customStyle }}
          block={block}
          disabled={disabled}
          type={type}
          onClick={onClick}
          data-testid="btn">
          {children}
        </RButton>
      )}
      {variant && (
        <RButton
          variant={variant}
          style={{ ...customStyle }}
          block={block}
          disabled={disabled}
          type={type}
          onClick={onClick}
          data-testid="btn">
          {children}
        </RButton>
      )}
    </>
  )
}
