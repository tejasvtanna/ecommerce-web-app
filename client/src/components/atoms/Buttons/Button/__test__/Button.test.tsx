import React from 'react'
import { Button } from '../Button'
import { fireEvent, render, screen, cleanup } from '@testing-library/react'
const renderer = require('react-test-renderer')

afterEach(cleanup)

// test # 1
test('<Button/> snapshot testing', () => {
  const component = renderer.create(
    <Button variant="light" block>
      First Button
    </Button>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// test # 2
test('Button is not disabled by default', () => {
  render(<Button></Button>)
  const btn = screen.getByTestId('btn')
  expect(btn).not.toBeDisabled()
})

// test # 3
test('Button is disabled', () => {
  render(<Button disabled={true}></Button>)
  const btn = screen.getByTestId('btn')
  expect(btn).toBeDisabled()
})

// test # 4
test('Button is not disabled', () => {
  render(<Button disabled={false}></Button>)
  const btn = screen.getByTestId('btn')
  expect(btn).not.toBeDisabled()
})

// test # 5
test('Callback is fired', () => {
  const onClickMock = jest.fn()
  render(<Button onClick={onClickMock} />)

  fireEvent.click(screen.getByTestId('btn'))
  expect(onClickMock).toHaveBeenCalled()
})
