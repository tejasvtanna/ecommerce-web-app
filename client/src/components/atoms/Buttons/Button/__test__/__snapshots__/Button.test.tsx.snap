// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> snapshot testing 1`] = `
<div>
  <button
    className="btn btn-light btn-block"
    data-testid="btn"
    disabled={false}
    style={Object {}}
    type="button"
  >
    First Button
  </button>
</div>
`;
